src/Engine/GameState/I_GameState.h

```diff
@@ -10,9 +10,26 @@ namespace gamestates {
     };
 }
 
+/**
+ * Many objects are intended to inherit I_GameState.
+ * This is the top level type in the Terminal Fighter Engine for handling game flow.
+ * This interface allows for GameStateHandler to manage various GameStates.
```

Comment:
Upon reading this description... I know it says `states` a lot, but I think it'd be beneficial to mention that this is supposed to work as a state machine.